server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -130,7 +135,7 @@
  * @opensearch.api
  */
 @PublicApi(since = "1.0.0")
-public abstract class Engine implements LifecycleAware, Closeable {
+public abstract class Engine implements LifecycleAware, Closeable, Indexer, CheckpointState, StatsHolder, IndexingThrottler {
```

Comment:
This might be a breaking change for this class espl when we can have alternate implementations governed by `EnginePlugin`